File: front-end/src/app/subscriber-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Subscriber } from './subscriber';



@Injectable({
  providedIn: 'root',
})
export class SubscriberService {
  private baseURL = "http://localhost:8080/ims/subscriber"

  constructor(private httpClient: HttpClient) {}

  getSubscribersList(): Observable<Subscriber[]>{
    return this.httpClient.get<Subscriber[]>(`${this.baseURL}`);     
  }

  createSubscriber(subscriber: Subscriber): Observable<Object>{
    return this.httpClient.post(`${this.baseURL}`, subscriber);
  }

  getSubscriberByphoneNumber(phoneNumber: number): Observable<Subscriber>{
     return this.httpClient.get<Subscriber>(`${this.baseURL}/${phoneNumber}`);
  }

   updateSubscriber(phoneNumber: number, subscriber: Subscriber): Observable<Object>{
    return this.httpClient.put(`${this.baseURL}/${phoneNumber}`, subscriber);
  }

  deleteSubscriber(phoneNumber: number): Observable<Object>{
    return this.httpClient.delete(`${this.baseURL}/${phoneNumber}`);
  }

}

File: springboot-ims/src/main/java/com/example/demo/service/SubscriberService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.dto.SubscriberDto;

public interface SubscriberService {
	
	SubscriberDto createSubscriber(SubscriberDto subscriberDto);
	
	SubscriberDto getSubscriberByPhoneNumber(Long phoneNumber);
	
	List<SubscriberDto> getAllSubscribers();
	
	SubscriberDto updateSubscriber(Long phoneNumber, SubscriberDto updateSubscriber);

	void deleteSubscriber(Long phoneNumber);
}
